src/uthash.h

```diff
@@ -573,18 +602,22 @@ do {
 /* The Paul Hsieh hash function */
 #undef get16bits
 #if (defined(__GNUC__) && defined(__i386__)) || defined(__WATCOMC__)             \
-  || defined(_MSC_VER) || defined (__BORLANDC__) || defined (__TURBOC__)
-#define get16bits(d) (*((const uint16_t *) (d)))
+  || defined(_MSC_VER) || defined (__BORLANDC__) || defined (__TURBOC__)         \
+  || defined(uthash_uint16_t)
+#ifndef uthash_uint16_t
+#define uthash_uint16_t uint16_t
+#endif
+#define get16bits(d) (*((const uthash_uint16_t *) (d)))
 #endif
```

Comment:
I thought for a while on what needs to happen here. Apparently, not using standard includes doesn't define __I386__ so on Linux, this path is not taken. I figure if the caller defined uthash_uint16_t, we should always use that, hence the update.